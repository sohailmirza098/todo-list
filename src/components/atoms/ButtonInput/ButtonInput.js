// import React from 'react'

// export default function ButtonInput() {
//     return (
//         <input data-testid="submittask" type="submit" value="Submit" className="btn" style={{flex: '1'}} />
//     )
// }


import React from 'react'

const ButtonInput = (props) => (
    <input data-testid="submittask" type="submit" value="Submit" className="btn" style={{flex: '1'}} />
)

export default ButtonInput; 