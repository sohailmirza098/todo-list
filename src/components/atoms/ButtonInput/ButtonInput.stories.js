import React from 'react';
import ButtonInput from "./ButtonInput";
import { action } from '@storybook/addon-actions';

export default {
  title: 'ButtonInput',
  component: ButtonInput,
};

const buttonData = {
  hidden: 'false',
  setHidden: action('Button Clicked')
}


export const ButtonInputStory = () => (
  <ButtonInput />
);

// npx -p @storybook/cli sb init --type react_scripts