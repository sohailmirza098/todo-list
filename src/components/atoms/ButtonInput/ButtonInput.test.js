import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ButtonInput from "./ButtonInput";

describe("Input Button testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<ButtonInput />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Input is present", () => {
    const {getByTestId, getByText} =  render(<ButtonInput />);
    const button = getByTestId("submittask");
    expect(button).toBeDefined();
  });

//   test("RemoveButton has content X", () => {
//     const {getByRole}=  render(<RemoveButton />);
//     const button = getByRole("button");
//     expect(button).toHaveTextContent("x");
//   });

//   test("RemoveButton onclick",()=>{
//     const mockClickMethod = jest.fn();
//     const {getByRole}=  render(<RemoveButton onClick={mockClickMethod()}/>);
//     fireEvent.click(getByRole("button"));
//     expect(mockClickMethod).toHaveBeenCalledTimes(1);
//   });
});