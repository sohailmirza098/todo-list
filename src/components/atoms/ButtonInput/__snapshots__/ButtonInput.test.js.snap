exports[`Input Button testing Matches the snapshot 1`] = `
<input
  className="btn"
  data-testid="submittask"
  style={
    Object {
      "flex": "1",
    }
  }
  type="submit"
  value="Submit" />
`;
